Utils/Services/ImagePickerService.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ImagePickerService.swift
+//  Utils
+//
+//  Created by Nahuel Gladstein on 6/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import ReactiveSwift
+import UIKit
+import AVFoundation
+import Photos
+import Result
+
+public enum ImagePickerServiceError: Error {
+    
+    case sourceTypeNotAvailable
+    
+}
+
+public protocol ImagePickerServiceType {
```

Comment:
We could use the internal `UIImagePickerSourceType` enum directly for that. 
About receiving the controller in every method I'm not sure, we could do that but as you will probably use the same controller it may be better to leave it in the `init`